### PR TITLE
Fixes the mini checklist from 19/02/2020:

### PR DESCRIPTION
**Pull request by Kmc2000**
## About The Pull Request

What's in this:
Fighter bugfixes such as:
Crashing into the ship
Not being able to get into fighters
Fighters drinking wayyy too much fuel
Syndicate fighters being too strong

Assistant uniform resprite.

Re-adds syndicate boarders and bug fixes them.
New explosion visuals courtesy of goonstation 2020

Syndie fighters receive an AI upgrade:
https://streamable.com/bqw9s

## Changelog
:cl: Kmc2000
fix: Nerfed syndicate fighters and fixed some issues with fighters
/:cl:

### CODE REVIEW

File: nsv13/code/modules/overmap/effects.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//All credit goes to Goonstation's 2020 release for these explosion sprites, thanks goons!//
+
+/obj/effect/temp_visual/impact_effect/torpedo
+	icon = 'nsv13/goonstation/icons/effects/explosions/60x60.dmi'
+	icon_state = "explosion"
+	duration = 2 SECONDS
+
+/obj/effect/temp_visual/impact_effect/torpedo/Initialize()
+	icon_state = "[prob(50) ? "explosion" : "explosion2"]"
```
Comment (francinum) [inline nsv13/code/modules/overmap/effects.dm]: This should probably be pick()d from a static list.

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 2]
```diff
@@ -770,7 +754,7 @@ After going through this checklist, you're ready to go!
 		to_chat(user, "<span class='warning'>[src]'s canopy isn't open.</span>")
 		return
 	if(maint_state < MS_UNSECURE)
-		if(alert("Enter what seat?",name,"Pilot seat","Passenger seat") !="Passenger seat")
+		if(alert("Enter what seat?",name,"Pilot seat","Passenger seat") == "Pilot seat")
```
Comment (francinum) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: These could probably be defines.

File: nsv13/code/modules/overmap/ai.dm [hunk 3]
```diff
@@ -73,10 +73,19 @@
 		if(get_dist(ship, src) <= 3)
 			user_thrust_dir = 0 //Don't thrust towards ships we're already close to.
 			brakes = TRUE
+			try_board(ship)
 		else
 			user_thrust_dir = 1
 			brakes = FALSE
 
+/obj/structure/overmap/proc/try_board(obj/structure/overmap/ship)
+	if(mass <= 1)
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/ai.dm]: Would be better to use the define here for readability

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 4]
```diff
@@ -770,7 +777,7 @@ After going through this checklist, you're ready to go!
 		to_chat(user, "<span class='warning'>[src]'s canopy isn't open.</span>")
 		return
 	if(maint_state < MS_UNSECURE)
-		if(max_passengers > 0 && alert("Enter what seat?",name,"Pilot seat","Passenger seat") !="Passenger seat")
+		if(alert("Enter what seat?",name,"Pilot seat","Passenger seat") == "Pilot seat")
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: A cancel button would also be nice.

**Comment by Vasily2013** [comment]
- [ ] Captain's office has no blast doors under windows.
- [ ] Captain's office walls are not reinforced.
- [ ] AI core doors should not start bolted
- [ ] XO office walls should be reinforced
- [ ] Put a firelock under the ORM and Requests consoles to prevent fire spread when the windows break.

**Comment by Cenrus** [comment]
> 
> 
>     * [ ]  Captain's office has no blast doors under windows.
> 
>     * [ ]  Captain's office walls are not reinforced.
> 
>     * [ ]  AI core doors should not start bolted
> 
>     * [ ]  XO office walls should be reinforced
> 
>     * [ ]  Put a firelock under the ORM and Requests consoles to prevent fire spread when the windows break.

The Enterprise was only altered in this PR to fix the broken tiles, I can commit these changes to my Enterprise PR if you'd like.

**Comment by Vasily2013** [comment]
> The Enterprise was only altered in this PR to fix the broken tiles, I can commit these changes to my Enterprise PR if you'd like.

Go ahead and do so.

File: nsv13/code/controllers/subsystem/processing/overmap.dm [hunk 5]
```diff
@@ -5,3 +5,4 @@ PROCESSING_SUBSYSTEM_DEF(overmap)
 	wait = 2
 	stat_tag = "OP"
 	priority = FIRE_PRIORITY_OVERMAP
+	var/next_boarding_time = 0
```
Comment (covertcorvid) [inline nsv13/code/controllers/subsystem/processing/overmap.dm]: Tell me this doesn't make boarding happen immediately.

File: nsv13/code/modules/overmap/ai_interiors.dm [hunk 6]
```diff
@@ -146,7 +147,6 @@ GLOBAL_LIST_INIT(drop_trooper_teams, list("Noble", "Helljumper","Red", "Black",
 		message_admins("[ADMIN_LOOKUPFLW(H)] became a syndicate drop trooper.")
 		to_chat(H, "<span class='danger'>You are a syndicate drop trooper! Your mission is to sabotage [station_name()] so that our ships can dispose of it. For Abassi!")
 		H.mind?.special_role = "Syndicate Drop Trooper"
-
+//	new /obj/effect/DPtarget(target, toLaunch)
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/ai_interiors.dm]: If this code isn't used it should be removed

File: nsv13/code/modules/overmap/effects.dm (hunk 1 above)
Comment (covertcorvid) [reply nsv13/code/modules/overmap/effects.dm]: Agree. Wouldn't prob(50) almost always be true since the default behavior is anything that's not null or 0 is true?

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 7]
```diff
@@ -471,7 +465,7 @@ After going through this checklist, you're ready to go!
 /obj/structure/overmap/fighter/slowprocess()
 	. = ..()
 	use_fuel()
-	if(canopy_breached || (canopy_open && pilot)) //Leak.
+	if(canopy_breached) //Leak air if the canopy is breached.
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: I think the ideal solution would be to average the air if the canopy was breached or open on a non-overmap Z, and just remove it on the overmap Z because no atmos. Agree that it didn't make sense to have air behavior depend on the presence of a pilot though.

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 8]
```diff
@@ -670,6 +664,19 @@ After going through this checklist, you're ready to go!
 		else
 			to_chat(user, "<span class='notice'>You require [src] to be in maintenance mode to load munitions!.</span>")
 			return
+	if(istype(A, /obj/structure/overmap/fighter/prebuilt/escapepod) && ispath(has_escape_pod) && (!escape_pod || escape_pod?.loc != src))
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: has_escape_pod sounds like it should be a boolean, not a path?
what is the point of has_escape_pod if there is also escape_pod?

File: nsv13/code/modules/overmap/fighters/fighters.dm (hunk 4 above)
Comment (covertcorvid) [reply nsv13/code/modules/overmap/fighters/fighters.dm]: Actually, doesn't this still give both options even for fighters without a passenger seat?

File: nsv13/code/controllers/subsystem/processing/overmap.dm (hunk 5 above)
Comment (Kmc2000) [reply nsv13/code/controllers/subsystem/processing/overmap.dm]: Intentional. But this only happens when large enemies get up close so it's already rare.

File: nsv13/code/modules/overmap/fighters/fighters.dm (hunk 4 above)
Comment (Jalleo) [reply nsv13/code/modules/overmap/fighters/fighters.dm]: It does

File: nsv13/code/controllers/subsystem/processing/overmap.dm (hunk 5 above)
Comment (Jalleo) [reply nsv13/code/controllers/subsystem/processing/overmap.dm]: Should be a warning and a minimum of ten seconds.

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 9]
```diff
@@ -65,6 +65,14 @@ After going through this checklist, you're ready to go!
 	var/warmup_cooldown = FALSE //So you cant blitz the fighter ignition in 2 seconds
 	var/ejecting = FALSE
 	var/throttle_lock = FALSE
+	var/has_escape_pod = /obj/structure/overmap/fighter/prebuilt/escapepod
```
Comment (Jalleo) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: We will need to rewrite the smaller craft as a lot of it ends up being hacky code to keep it together. That is out of scope for this PR.
Escapepods cant be children of fighters

File: nsv13/code/modules/overmap/fighters/fighters.dm [hunk 10]
```diff
@@ -983,23 +990,31 @@ After going through this checklist, you're ready to go!
...
-	if(operators.len && !istype(src, /obj/structure/overmap/fighter/prebuilt/escapepod))
+	if(operators.len && escape_pod && escape_pod.loc == src)
 		relay('nsv13/sound/effects/computer/alarm_3.ogg', "<span class=userdanger>EJECT! EJECT! EJECT!</span>")
 		relay_to_nearby('nsv13/sound/effects/ship/fighter_launch_short.ogg')
 		visible_message("<span class=userdanger>Auto-Ejection Sequence Enabled! Escape Pod Launched!</span>")
-		eject()
-		sleep(20)
+		ejecting = FALSE
+		if(eject())
+			sleep(20)
```
Comment (Jalleo) [inline nsv13/code/modules/overmap/fighters/fighters.dm]: Timer no sleep!

File: nsv13/code/modules/clothing/custom_clothes.dm [hunk 11]
```diff
@@ -44,11 +44,25 @@
...
 	name = "Placeholder"
 	icon = 'nsv13/icons/obj/clothing/suits.dmi' //Placeholder subtype for our own iconsets
 	alternate_worn_icon = 'nsv13/icons/mob/suit.dmi'
 
+/obj/item/clothing/suit/ship/assistant_jacket
+	name = "Utility jacket"
+	icon_state = "assistant_jacket"
+	item_color = "assistant_jacket"
+	desc = "A stylish but heavy jacket designed to protect its wearer from common hazards when serving aboard a ship."
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 10, "bio" = 0, "rad" = 10, "fire" = 10, "acid" = 10)
```
[the remark this answers is not in the document]
Comment (Karmic-Skink) [reply nsv13/code/modules/clothing/custom_clothes.dm]: Wot.

[the remark this answers is not in the document]
Comment (Jalleo) [reply nsv13/code/modules/clothing/custom_clothes.dm]: Assistants should be given dangerous to wear jackets since they are cannon fodder.
